fsspec/mapping.py

```diff
@@ -61,28 +84,44 @@ def __init__(self, root, fs, check=False, create=False, missing_exceptions=None)
             self.fs.rm(root + "/a")
 
     @cached_property
-    def dirfs(self):
+    def dirfs(self) -> DirFileSystem:
         """dirfs instance that can be used with the same keys as the mapper"""
         from .implementations.dirfs import DirFileSystem
 
         return DirFileSystem(path=self._root_key_to_str, fs=self.fs)
 
-    def clear(self):
+    def clear(self) -> None:
         """Remove all keys below root - empties out mapping"""
         try:
             self.fs.rm(self.root, True)
             self.fs.mkdir(self.root)
         except:  # noqa: E722
             pass
 
-    def getitems(self, keys, on_error="raise"):
+    @overload
```

Comment:
This is a bit of ugliness to make the output type of `getitems` dynamic based on `on_error`

Reply:
Can we not just use `str` and ignore the overload? I don't think this kind of construct is worth it.